### PR TITLE
Revive the marine jcb implementation

### PR DESCRIPTION
**Pull request by guillaumevernieres**
Some updates/fixes to the marine yaml templates.

### CODE REVIEW

File: algorithm/marine/socaincr2mom6.yaml.j2 [hunk 1]
```diff
@@ -2,27 +2,27 @@ geometry:
...
 vertical geometry:
-  date: '{{marine_window_begin}}'
+  date: '{{ MARINE_WINDOW_BEGIN_ISO }}'
   basename: ./INPUT/
   ocn_filename: MOM.res.nc
   read_from_file: 1
 
 soca increment:
-  date: '{{marine_window_begin}}'
+  date: '{{ MARINE_WINDOW_BEGIN_ISO }}'
```
Comment (CoryMartin-NOAA) [inline algorithm/marine/socaincr2mom6.yaml.j2]: will defer to JCB's father @danholdaway on style guidance but I think most of these JCB algorithm templates would look more like `{{ marine_window_begin_iso }}`

Comment (danholdaway) [reply algorithm/marine/socaincr2mom6.yaml.j2]: Why are you YELLING at us G?

Comment (guillaumevernieres) [reply algorithm/marine/socaincr2mom6.yaml.j2]: # **I'M NOT YELLING.**

**Comment by guillaumevernieres** [comment]
As I'm adjusting the cases, I'm realizing that I'm not making use of the soca_2cice yaml ... Is that the proper location for these yamls?
- soca_2cice: it's coded as a change of variables and inserts the ice analysis in the cice restart
- the soca2mom6 yaml is for the app that prepares the mom6 iau increment

**Review by AndrewEichmann-NOAA — COMMENTED** [review]
looks ok to me

File: algorithm/marine/socaincr2mom6.yaml.j2 (hunk 1 above)
Comment (danholdaway) [reply algorithm/marine/socaincr2mom6.yaml.j2]: > # **I'M NOT YELLING.**

Doesn't mixing cases add unnecessary confusion and back and forth?

You have to prescribe a mapping between G-W and JCB in GDASapp anyway so I don't think it's all the complicated to have singular case on the jcb side.

Comment (guillaumevernieres) [reply algorithm/marine/socaincr2mom6.yaml.j2]: lower-cased stuff in my last commit.

**Review by danholdaway — APPROVED** [review]

**Comment by CoryMartin-NOAA** [comment]
@danholdaway just an FYI I think the JCB-GDAS CI tests always fail, not sure why (haven't looked into it)

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**